subsys/bluetooth/mesh/brg_cfg_cli.c

```diff
@@ -31,47 +151,170 @@ const struct bt_mesh_model_cb _bt_mesh_brg_cfg_cli_cb = {
+	struct bt_mesh_msg_ctx ctx = BT_MESH_MSG_CTX_INIT_DEV(net_idx, addr);
+	const struct bt_mesh_msg_rsp_ctx rsp_ctx = {
+		.ack = &cli->ack_ctx,
+		.op = OP_SUBNET_BRIDGE_STATUS,
+		.user_data = status,
+		.timeout = msg_timeout,
+	};
+
+	bt_mesh_model_msg_init(&msg, OP_SUBNET_BRIDGE_GET);
+
+	return bt_mesh_msg_ackd_send(cli->model, &ctx, &msg, !status ? NULL : &rsp_ctx);
 }
 
 int bt_mesh_brg_cfg_cli_subnet_bridge_set(uint16_t net_idx, uint16_t addr,
 					  enum bt_mesh_subnet_bridge_state val,
 					  enum bt_mesh_subnet_bridge_state *status)
 {
-	return 0;
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_SUBNET_BRIDGE_SET, 1);
+	struct bt_mesh_msg_ctx ctx = BT_MESH_MSG_CTX_INIT_DEV(net_idx, addr);
+	const struct bt_mesh_msg_rsp_ctx rsp_ctx = {
+		.ack = &cli->ack_ctx,
+		.op = OP_SUBNET_BRIDGE_STATUS,
+		.user_data = status,
+		.timeout = msg_timeout,
+	};
+
+	bt_mesh_model_msg_init(&msg, OP_SUBNET_BRIDGE_SET);
+	net_buf_simple_add_u8(&msg, (uint8_t)val);
+
+	return bt_mesh_msg_ackd_send(cli->model, &ctx, &msg, !status ? NULL : &rsp_ctx);
 }
 
 int bt_mesh_brg_cfg_cli_bridging_table_size_get(uint16_t net_idx, uint16_t addr, uint16_t *size)
 {
-	return 0;
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_BRIDGING_TABLE_SIZE_GET, 0);
+	struct bt_mesh_msg_ctx ctx = BT_MESH_MSG_CTX_INIT_DEV(net_idx, addr);
+	const struct bt_mesh_msg_rsp_ctx rsp_ctx = {
+		.ack = &cli->ack_ctx,
+		.op = OP_BRIDGING_TABLE_SIZE_STATUS,
+		.user_data = size,
+		.timeout = msg_timeout,
+	};
+
+	bt_mesh_model_msg_init(&msg, OP_BRIDGING_TABLE_SIZE_GET);
+
+	return bt_mesh_msg_ackd_send(cli->model, &ctx, &msg, !size ? NULL : &rsp_ctx);
 }
 
 int bt_mesh_brg_cfg_cli_bridging_table_add(uint16_t net_idx, uint16_t addr, uint8_t directions,
 					   uint16_t net_idx1, uint16_t net_idx2,
 					   uint16_t addr1, uint16_t addr2,
 					   struct bt_mesh_bridging_table_status *rsp)
 {
-	return 0;
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_BRIDGING_TABLE_ADD, 8);
+	struct bt_mesh_msg_ctx ctx = BT_MESH_MSG_CTX_INIT_DEV(net_idx, addr);
+	const struct bt_mesh_msg_rsp_ctx rsp_ctx = {
+		.ack = &cli->ack_ctx,
+		.op = OP_BRIDGING_TABLE_STATUS,
+		.user_data = rsp,
+		.timeout = msg_timeout,
+	};
+
+	if (addr1 == addr2) {
+		LOG_ERR("addr1 and addr2 shall have different values.");
+		return -EINVAL;
+	} else if (!BT_MESH_ADDR_IS_UNICAST(addr1)) {
+		LOG_ERR("addr1 shall be a unicast address.");
+		return -EINVAL;
+	} else if (directions == 0x01 &&
+		   (addr2 == BT_MESH_ADDR_UNASSIGNED || addr2 == BT_MESH_ADDR_ALL_NODES)) {
+		LOG_ERR("For direction 0x01: addr2 shall not be unassigned or the all-nodes fixed "
+			"group address.");
+		return -EINVAL;
+	} else if (directions == 0x02 && !BT_MESH_ADDR_IS_UNICAST(addr2)) {
+		LOG_ERR("For direction 0x02: addr2 shall be a unicast address.");
+		return -EINVAL;
+	}
+
+	bt_mesh_model_msg_init(&msg, OP_BRIDGING_TABLE_ADD);
+	net_buf_simple_add_u8(&msg, directions);
+	net_buf_simple_add_le24(&msg, (net_idx1 | (net_idx2 << 12)));
```

Comment:
net_idx1/2 is of type uint16_t, but `net_buf_simple_add_le24` expects a uint32_t. Is there a problem with inputting the argument as it is here?

I did some local testing and it seems the uint16_t are "promoted" such that there is no overflow, but we may want to explicitly cast them to uint32_t before the shifting operation?

Reply:
I did minor prototyping in online gcc. Seems it works. It will be nice to get confirmation from C standard. I know the width of variables are adjusted to max type in equation, but I'm not sure from which stage.

Reply:
(Discussed elsewhere): According to C99 Integer promotion is performed in bitwise shift operations. local testing indicates that this should work.